### PR TITLE
travis: add cache configuration for $GOPATH/{src,pkg}

### PR DESCRIPTION
**Pull request by broady**

### CODE REVIEW

**Comment by broady** [comment]
Should speed up tests by about 40 seconds.

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,15 @@ go:
 - 1.6
 #- tip # NOTE: disabled - see https://github.com/travis-ci/gimme/issues/38
 
+before_cache:
+  - rm -rf $GOPATH/src/github.com/GoogleCloudPlatform/golang-samples
```
Comment (adg) [inline .travis.yml]: why is there a space before these bullets, but not down by 'install:'

**Comment by adg** [comment]
Happy for you to merge this.

File: .travis.yml (hunk 1 above)
Comment (broady) [inline .travis.yml]: I'm not sure. To be honest, I don't really understand the YAML syntax. I'll try removing them.

**Pull request merged**